launch_ros/launch_ros/utilities/namespace_utils.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module with utility functions for handling namespaces."""
+
+
+from typing import Optional
+from typing import Text
+
+
+def is_root_namespace(ns: Text) -> bool:
+    """Return `True` if `ns` is `'/'`."""
+    return ns == '/'
+
+
+def is_namespace_absolute(ns: Text) -> bool:
+    """Return `True` if `ns` is absolute."""
+    return ns.startswith('/')
+
+
+def prefix_namespace(
+    base_ns: Optional[Text],
+    ns: Optional[Text],
+    *,
+    return_absolute_ns: bool = True,
+) -> Optional[Text]:
+    """
+    Return `ns` prefixed with `base_ns` if `ns` is relative, returns `ns` if not.
+
+    :param `base_ns`: Prefix to be added to `ns`.
+    :param `ns`: Namespace to be prefixed.
+    :param `return_absolute_ns`: if `True`, the result will be made absolute.
```

Comment:
@ivanpauno meta: instead of having this argument, consider using `make_absolute_namespace` at `prefix_namespace` call site when necessary e.g.:

```python
make_absolute_namespace(prefix_namespace(base_ns, ns))
```

Reply:
I thought about that, the problem is that I actually need:
```python3
combined_ns = prefix_namespace(base_ns, ns)
if combined_ns is not None:
  combined_ns = make_absolute_namespace(combined_ns)
```

and I was too lazy too write all that :joy:.

I can also modify `make_absolute_namespace` to handle `None` inputs, but I preferred not accepting that.
WDYT?

Reply:
Hmm, since `prefix_namespace` does handle `None`, changing `make_absolute_name` to do so as well doesn't sound bad. But I don't mind to leave as-is.

Reply:
Maybe remove all `return_absolute_ns` references from the documentation?

Reply:
thanks for catching that

Reply:
Fixed in https://github.com/ros2/launch_ros/pull/162/commits/5c22c0f8262b8b970c5417063124d2ddccac8d91.